src/snapshot.js

```diff
@@ -93,6 +117,15 @@ export function restorePlayerSnapshot(player, snapshotObject) {
 
   const suppressedRemoteTracks = snapshotObject.suppressedRemoteTracks;
   const suppressedTracks = snapshotObject.suppressedTracks;
+
+  // ensures that the iOS TextTrackList 'change' listener added in getPlayerSnapshot()
+  // doesn't persist into main content
+  if (videojs.browser.IS_IOS) {
```

Comment:
I suppose you would want to specify `featuresNativeTextTracks` here too if we do in getPlayerSnapshot